pkgs/npm/src/cli.ts

```diff
@@ -24,14 +24,15 @@ function getPathToExecutable(): string {
 // in the user PATH to the cli.js to execute. The symlink will name the same as
 // the package name i.e. defang.
 function run(): void {
+  let processResult: child_process.SpawnSyncReturns<Buffer> | null = null;
   try {
     const args = process.argv.slice(2);
-    const processResult = child_process.spawnSync(getPathToExecutable(), args, {
+    processResult = child_process.spawnSync(getPathToExecutable(), args, {
       stdio: "inherit",
     });
   } catch (error) {
     console.error(error);
-    process.exit(1);
+    process.exitCode = processResult?.status ?? 1;
```

Comment:
`processResult` will always be `null` since it's `spawnSync` that throws before the assignment takes place. Did you mean to pull `process.exitCode = …` outside of the catch block?

Reply:
IIRC, TS will be smart enough to figure out the type if you just have `let processResult;`